tests/test_multitiffmultipageextractor.py

```diff
@@ -0,0 +1,232 @@
+        """Test retrieving frames from the extractor."""
+        file_paths = sorted(list(self.folder_path.glob("*.tif")))
+
+        # Initialize with XYZCT dimension order
+        extractor = MultiTIFFMultiPageExtractor(
+            file_paths=file_paths,
+            dimension_order="XYZCT",
+            channel_size=1,
+            depth_size=3,
+            time_size=2,
+            sampling_frequency=30.0,
+        )
+
+        # Get specific frames
+        frame_0 = extractor.get_frames(0)
+        frame_1 = extractor.get_frames(1)
+
+        # Check frame shapes - should include depth dimension
+        self.assertEqual(frame_0.shape, (10, 12, 3))
+        self.assertEqual(frame_1.shape, (10, 12, 3))
+
+        # Get multiple frames
+        frames = extractor.get_frames([0, 1])
+        self.assertEqual(frames.shape, (2, 10, 12, 3))
+
+    def test_get_video(self):
+        """Test retrieving video from the extractor."""
+        file_paths = sorted(list(self.folder_path.glob("*.tif")))
+
+        # Initialize with XYZCT dimension order
+        extractor = MultiTIFFMultiPageExtractor(
+            file_paths=file_paths,
+            dimension_order="XYZCT",
+            channel_size=1,
+            depth_size=3,
+            time_size=2,
+            sampling_frequency=30.0,
+        )
+
+        # Get full video
+        video = extractor.get_video()
+        self.assertEqual(video.shape, (2, 10, 12, 3))
+
+        # Get partial video
+        partial_video = extractor.get_video(start_frame=0, end_frame=1)
+        self.assertEqual(partial_video.shape, (1, 10, 12, 3))
+
+    def test_from_folder(self):
+        """Test creating an extractor from a folder path."""
+        # Create extractor using from_folder
+        extractor = MultiTIFFMultiPageExtractor.from_folder(
+            folder_path=self.folder_path,
+            file_pattern="*.tif",
+            dimension_order="XYZCT",
+            channel_size=1,
+            depth_size=3,
+            time_size=2,
+            sampling_frequency=30.0,
+        )
+
+        # Check basic properties
+        self.assertEqual(extractor.get_num_frames(), 2)
+        self.assertEqual(extractor.get_image_size(), (10, 12))
+
+    def test_different_dimension_orders(self):
+        """Test different dimension orders."""
+        file_paths = sorted(list(self.folder_path.glob("*.tif")))
+
+        # Test with XYZTC dimension order
+        extractor_xyztc = MultiTIFFMultiPageExtractor(
+            file_paths=file_paths,
+            dimension_order="XYZTC",
+            channel_size=1,
+            depth_size=3,
+            time_size=2,
+            sampling_frequency=30.0,
+        )
+
+        # Check basic properties
+        self.assertEqual(extractor_xyztc.get_num_frames(), 2)
+
+        # Get frames and check they're accessible
+        frame_0 = extractor_xyztc.get_frames(0)
+        self.assertEqual(frame_0.shape, (10, 12, 3))
+
+    def test_comparison_with_scanimage(self):
```

Comment:
Review this. This is the most critical part. Apparently the scan image example file used here should not be volumetric.